includes/class-sensei-course.php

```diff
@@ -145,17 +145,81 @@ public function __construct() {
 
 		add_action( 'template_redirect', [ $this, 'setup_single_course_page' ] );
 		add_action( 'sensei_loaded', [ $this, 'add_legacy_course_hooks' ] );
+
+		// Add custom navigation.
+		add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
+		add_action( 'parent_file', [ $this, 'highlight_menu_item' ] );
 	}
 
 	/**
-	 * Register and enqueue scripts that are needed in the backend.
+	 * Highlight the menu item for the course pages.
+	 *
+	 * @param string $file
+	 *
+	 * @return string
+	 */
+	public function highlight_menu_item( $file ) {
+		global $submenu_file;
+		$screen = get_current_screen();
+
+		if ( $screen && in_array( $screen->id, [ 'edit-course', 'edit-course-category', 'course_page_course-order' ], true ) ) {
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			$submenu_file = 'edit.php?post_type=course';
+		}
+
+		return $file;
```

Comment:
I've noticed you're calling a filter as action, so I'm not sure if you need to do a return in this case.

Reply:
Actually, I'm not calling it. I just attach my filter to the hook, and it is supposed to be a filter.

Reply:
I think it's ok to use a filter as an action if you are not changing the input.

Looking at the code now, you are updating the `$submenu_file`. Isn't it better to use the `submenu_file` filter instead?

Reply:
Definitely, this filter is much-much better! Thanks!

Reply:
Fixed here: https://github.com/Automattic/sensei/pull/4672/commits/e5f525110cc9d3e2e4a8dc26d9e3b36108d6a5b8